app/routes/datasets/versions.py

```diff
@@ -74,6 +86,12 @@ async def add_new_version(
     """Create or update a version for a given dataset."""
 
     input_data = request.dict(exclude_none=True, by_alias=True)
+
+    creation_options = input_data["creation_options"]
+    del input_data["creation_options"]
```

Comment:
You could also do
creation_options = input_data.pop("creation_options")
It's no better, but it's only one line. :)

Reply:
good point